python/dgl/transform.py

```diff
@@ -1991,13 +2000,32 @@ def to_simple(g,
         node_frames = utils.extract_node_subframes(g, None)
         utils.set_new_frames(simple_graph, node_frames=node_frames)
     if copy_edata:
-        eids = []
-        for i in range(len(g.canonical_etypes)):
-            feat_idx = F.asnumpy(edge_maps[i])
-            _, indices = np.unique(feat_idx, return_index=True)
-            eids.append(F.zerocopy_from_numpy(indices))
-
-        edge_frames = utils.extract_edge_subframes(g, eids)
+        if aggregator == 'random':
+            eids = []
+            for i in range(len(g.canonical_etypes)):
+                feat_idx = F.asnumpy(edge_maps[i])
+                _, indices = np.unique(feat_idx, return_index=True)
+                eids.append(F.zerocopy_from_numpy(indices))
+
+            edge_frames = utils.extract_edge_subframes(g, eids)
+        else:  # 'sum'/'mean'
+            edge_frames = []
+            for i in range(len(g.canonical_etypes)):
+                feat_idx = edge_maps[i]
+                _, indices = np.unique(F.asnumpy(feat_idx), return_index=True)
+                _num_rows = len(indices)
+                _data = {}
+                for key, col in g._edge_frames[i]._columns.items():
+                    data = col.data
+                    new_data = F.scatter_add(data, feat_idx, _num_rows)
```

Comment:
How does this compare to `torch.scatter_add_`?  I'm wondering if we should replace the usage of `scatter_add_` in GSpMM backward function if the current one is more efficient.

Reply:
This one is different from `torch.scatter_add_`, our `scatter_add` only requires the indices on the first dimension `(n,)` but torch's `scatter_add_` requires the indices on first two dimensions: `(n,d)`.

Reply:
We use torch's `scatter_add_` in GSpMM backward function because the argmax/min at each feature dimension is different.